File: sample/src/main/java/ir/farshid_roohi/customadapterrecyclerview/MyAdapter.java
package ir.farshid_roohi.customadapterrecyclerview;

import androidx.databinding.ViewDataBinding;

import ir.farshid_roohi.customadapterrecyclerview.databinding.MyItemBinding;
import ir.farshid_roohi.customadapterrecycleview.AdapterRecyclerView;
import ir.farshid_roohi.customadapterrecycleview.viewHolder.ItemViewHolder;

/**
 * Created by Farshid Roohi.
 * CustomAdapterRecyclerView | Copyrights 1/2/19.
 */
public class MyAdapter extends AdapterRecyclerView<String> {

    @Override
    public int getItemLayout(int viewType) {
        return R.layout.my_item;
    }

    // Optional method override to custom layout for endless progress
    @Override
    public int onProgressLayout() {
        return R.layout.my_custom_progress_item;
    }

    @Override
    public void onBindView(ViewDataBinding viewDataBinding, ItemViewHolder viewHolder, int position, int viewType, String element) {
        MyItemBinding itemBinding = (MyItemBinding) viewDataBinding;
        itemBinding.txtTitle.setText(element + " " + position);
    }
}

File: sample/src/main/java/ir/farshid_roohi/customadapterrecyclerview/MyAdapter.kt
package ir.farshid_roohi.customadapterrecyclerview

import android.content.Context
import ir.farshid_roohi.customadapterrecycleview.AdapterRecyclerView
import kotlinx.android.synthetic.main.my_item.view.*

/**
 * Created by Farshid Roohi.
 * CustomAdapterRecyclerView | Copyrights 1/2/19.
 */
class MyAdapter : AdapterRecyclerView<String?>(R.layout.my_item, R.layout.progress_view, R.layout.item_error,R.id.btnTrayAgain) {

    override fun onBindView(viewHolder: ItemViewHolder, position: Int, context: Context, element: String?) {
        val itemView = viewHolder.itemView
        itemView.txt_title.setText("$element $position")
    }

}